src/api/account.ts

```diff
@@ -874,4 +884,44 @@ export class Account {
   async deriveAccountFromPrivateKey(args: { privateKey: PrivateKey }): Promise<AccountModule> {
     return deriveAccountFromPrivateKey({ aptosConfig: this.config, ...args });
   }
+
+  /**
+   * Verifies a Secp256k1 account by checking the signature against the message and account's authentication key.
+   *
+   * This function takes a message and signature, and attempts to recover the public key that created the signature.
+   * It then verifies that the recovered public key matches the authentication key stored on-chain for the provided account address.
+   *
+   * If a recovery bit is provided, it will only attempt verification with that specific recovery bit.
+   * Otherwise, it will try all possible recovery bits (0-3) until it finds a match.
+   *
+   * @param args - The arguments for verifying the Secp256k1 account
+   * @param args.message - The message that was signed
+   * @param args.signature - The signature to verify (either raw hex or Secp256k1Signature object)
+   * @param args.recoveryBit - Optional specific recovery bit to use for verification
+   * @param args.accountAddress - The address of the account to verify
+   * @returns The recovered public key if verification succeeds
+   * @throws Error if verification fails or no matching public key is found
+   *
+   * @example
+   * ```typescript
+   * import { Aptos, AptosConfig, Network } from "@aptos-labs/ts-sdk";
+   *
+   * const config = new AptosConfig({ network: Network.DEVNET });
+   * const aptos = new Aptos(config);
+   *
+   * const publicKey = await aptos.verifySecp256k1Account({
+   *   message: "0x1234...",
+   *   signature: "0x5678...",
+   *   accountAddress: "0x1"
+   * });
+   * ```
+   */
+  async verifySecp256k1Account(args: {
+    message: HexInput;
+    signature: HexInput | Secp256k1Signature | AnySignature;
```

Comment:
why do we accept `AnySignature` if we only verify for `Secp256k1Account`? why is `Secp256k1Signature` is not part of `AnySignature`? Also, why do we accept `HexInput` as a signature? dont we want to make sure the function accepts a valid signature?

Reply:
I agree, public key recovery should be in the signature to return.  I don't think it belongs at the account level given you can't recover the private key

Reply:
Because Secp256k1Account does not exist.  SingleKeyAccount exists which returns AnySignature when it signs something.  You can get the inner Secp256k1Signature via signature.signature, but you would still have to type check it into Secp256k1Signature.  Having the function handle it seems better DevX

Reply:
@gregnazario not sure what you are suggesting

Reply:
Oh and HexInput is again for convenience.  The function will check signature validity.

Reply:
I also would expect this function to be in a different place than `Account`.

I think `Secp256k1PublicKey.fromMessageAndSignature` makes total sense (`fromSignedMessage` could be even more concise).

Ideally that's all the devs would need, and they could do the following:

```typescript
const innerPublicKey = new Secp256k1PublicKey.fromSignedMessage({ message, signature });
const publicKey = new AnyPublicKey({ publicKey: innerPublicKey });
const derivedAddress = publicKey.authKey().derivedAddress.toString();

// Simple case
const isValid = derivedAddress === accountAddress;

// Handle rotated auth keys
const { authentication_key } = await aptos.getAccountInfo({ accountAddress });
const isValid = derivedAddress === authentication_key;
```

now.. how can we make this even easier / more concise?
1) Going from `Secp256k1PublicKey` to `AnyPublicKey` is a bit annoying, maybe we can have the constructor there instead: `AnyPublicKey.fromSecp256k1SignedMessage({ message, signature })`
2) The action of verifying the publicKey is associated to an account from its address is agnostic to the signature scheme, so we could have `verifyAuthenticationKey({ authKey, accountAddress })` which could live in the `api` to avoid dependency cycles

Reply:
I really hate the `AnyPublicKey` name ugh

Reply:
So the problem is you need to look up the auth key with the address, and only via the auth key can you figure out which public key is correct.  If a recovery is provided 2 public keys would be returned.

I'm down for AnyPublicKey.fromSecp256k1SignedMessage({ message, signature }) though

Reply:
Yeah I re-evaluated it and did similar here https://github.com/aptos-labs/aptos-go-sdk/pull/108

Reply:
> So the problem is you need to look up the auth key with the address, and only via the auth key can you figure out which public key is correct. If a recovery is provided 2 public keys would be returned.

Oh I see, gotcha makes sense